frontend/src/employee-frontend/components/child-information/assistance-need/decision/assistance-need-decision-form.ts

```diff
@@ -8,15 +8,37 @@ import {
   AutosaveStatus,
   useAutosave
 } from 'employee-frontend/utils/use-autosave'
-import { AssistanceNeedDecision } from 'lib-common/generated/api-types/assistanceneed'
+import { Result } from 'lib-common/api'
+import {
+  AssistanceNeedDecision,
+  AssistanceNeedDecisionEmployee
+} from 'lib-common/generated/api-types/assistanceneed'
 import { UUID } from 'lib-common/types'
 
 import { getAssistanceNeedDecision, putAssistanceNeedDecision } from './api'
 
+type AssistanceNeedDecisionEmployeeForForm = Omit<
+  AssistanceNeedDecisionEmployee,
+  'name' | 'email'
+>
+
+export type AssistanceNeedDecisionForm = Omit<
+  AssistanceNeedDecision,
+  'selectedUnit' | 'preparedBy1' | 'preparedBy2' | 'decisionMaker'
+> & {
+  selectedUnit: { id: UUID | null } | null
+  preparedBy1: AssistanceNeedDecisionEmployeeForForm | null
+  preparedBy2: AssistanceNeedDecisionEmployeeForForm | null
+  decisionMaker: Omit<
+    AssistanceNeedDecisionEmployeeForForm,
+    'phoneNumber'
+  > | null
+}
```

Comment:
I think the `AssistanceNeedDecisionRequest` type should be updated in the controller and the generated result used in the frontend. Right now there's a bit of type confusion on the server as what the `GET` call returns is not what the `POST` or `PUT` calls use.